docs/src/rfc/text/0000-boot-manager.md

```diff
@@ -0,0 +1,707 @@
+   - **Security Variables**: TestKeyUsed (PCD-based warning display)
+   - **OS Communication**: OsIndications, OsIndicationsSupport
+
+3. **Protocol Consumption**:
+   - `EFI_SIMPLE_TEXT_INPUT_EX_PROTOCOL` - for hotkey support
+   - `EFI_DEFERRED_IMAGE_LOAD_PROTOCOL` - deferred image handling on ReadyToBoot
+   - `EDKII_VARIABLE_POLICY_PROTOCOL` - variable protection
+
+4. **Event Handling**:
+   - `gConnectConInEventGuid` - ConIn-on-demand support
+   - `gEfiEventReadyToBootGuid` - deferred image validation
+
+5. **PCD Dependencies**:
+   - Platform configuration: Timeout, language codes, firmware vendor/revision
+   - Feature flags: ConInConnectOnDemand, CapsuleOnDiskSupport, CapsuleFmpSupport, PlatformRecoverySupport
+   - Error handling: HardwareErrorRecordLevel
+
+6. **Platform Library Dependencies**:
+   - `PlatformBootManagerLib` - 4 callback functions (BeforeConsole, AfterConsole, WaitCallback, UnableToBoot)
+   - `UefiBootManagerLib` - boot option management utilities
+
+## Rust Code Design
+
+### Protocol Adapter for EDK2 Compatibility
+
+**Decision**: Implement an EDK2 protocol adapter layer that wraps Patina's service architecture.
+
+**Rationale**: For drop-in replacement, platforms must be able to remove BdsDxe.inf and add a Patina equivalent without
+modifying platform DSC files beyond the INF substitution. This requires:
+
+- Production of `EFI_BDS_ARCH_PROTOCOL` with correct GUID
+- DXE_DRIVER module type for DxeCore dispatch
+- Identical entry point signature and behavior
+- Compatible with existing PlatformBootManagerLib implementations
+
+**Implementation Strategy**:
+
+- Thin protocol adapter implements `EFI_BDS_ARCH_PROTOCOL::BdsEntry`
+- Adapter initializes Patina service registry
+- Registers default service implementations
+- Invokes Boot Orchestrator
+- Returns control to DxeCore upon boot failure (never returns on successful boot)
+
+**Trade-offs**: Small compatibility shim adds ~100 lines of glue code, but enables seamless EDK2 integration without
+requiring platforms to restructure their builds.
+
+### Configuration and Platform Dependencies
+
+**Decision**: Boot Manager consumes configuration and platform callbacks rather than managing dependencies.
+
+**Rationale**: EDK2's BdsDxe driver is a *consumer* of Platform Configuration Database (PCD) values and platform
+library callbacks, not a provider or dependency manager. This pattern must be preserved for drop-in replacement
+compatibility.
+
+**EDK2 Dependency Pattern**:
+
+BdsDxe **reads** PCDs at runtime:
+
+- `PcdFirmwareVendor`, `PcdFirmwareRevision` → System table initialization
+- `PcdPlatformBootTimeOut` → Boot timeout variable initialization
+- `PcdConInConnectOnDemand` → Console connection behavior control
+- `PcdPlatformRecoverySupport` → Recovery option enablement
+- `PcdCapsuleOnDiskSupport`, `PcdCapsuleFmpSupport` → Capsule feature flags
+
+BdsDxe **calls** platform library functions:
```

Comment:
I was going to suggest on the protocol dependencies, but my guess is that it will be just those arch protocols, we probably cannot do much better than that.

However, this `PlatformBootManager` abstract might contain a lot of content, and some are very much in common. i.e. the signal end of dxe, dispatch deferred images. Would consider redoing the abstraction with this RFC?